### PR TITLE
Fix: MAX17048 - Return NaN if no battery attached

### PR DESCRIPTION
**Pull request by tyeth**
This checks the version + ID before each read, returning NaN if not valid (no battery attached).

### CODE REVIEW

**Review by tyeth — COMMENTED** [review]
Tagging @brentru for review

**Comment by tyeth** [comment]
Not sure whats gone on with the build there, errors suggest some bracket isn't closed or idf changed or something is fubar.
Will look when I get back later

**Comment by brentru** [comment]
@tyeth  Run clang locally over `src/components/i2c/drivers/WipperSnapper_I2C_Driver_MAX17048.h`

File: src/components/i2c/drivers/WipperSnapper_I2C_Driver_MAX17048.h [hunk 1]
```diff
@@ -56,11 +56,28 @@ class WipperSnapper_I2C_Driver_MAX17048 : public WipperSnapper_I2C_Driver {
...
 
+  /*******************************************************************************/
+  /*!
+      @brief Checks the response of the MAX17048 chip by verifying ID and version.
+      @returns true if the chip ID or version is invalid, false otherwise.
+   */
+  /*******************************************************************************/
+  bool deviceNotReady() {
+    uint8_t chipID = _maxlipo->getChipID();
+    if (chipID == 0x00 || chipID == 0xFF)
```
Comment (brentru) [inline src/components/i2c/drivers/WipperSnapper_I2C_Driver_MAX17048.h:73]: Can you #define these numbers?

File: src/components/i2c/drivers/WipperSnapper_I2C_Driver_MAX17048.h [hunk 2]
```diff
@@ -56,11 +56,28 @@ class WipperSnapper_I2C_Driver_MAX17048 : public WipperSnapper_I2C_Driver {
...
+      return true;
+    uint16_t chipVersion = _maxlipo->getICversion();
+    if (chipVersion == 0x0000 || chipVersion == 0xFFFF)
```
Comment (brentru) [inline src/components/i2c/drivers/WipperSnapper_I2C_Driver_MAX17048.h:76]: Same here - can you check if they are present within the MAX library's header, and reference the macros or define them?

**Review by brentru — CHANGES_REQUESTED** [review]

**Comment by tyeth** [comment]
There are no longer changes required as the library has been augmented with returning NaNs, and the begin will fall if no battery attached (same check inside lib).

**Pull request closed without merging**